modules/blazingsql/src/blazingcluster.ts

```diff
@@ -0,0 +1,306 @@
+   * import {BlazingCluster} from '@rapidsai/blazingsql';
+   *
+   * const a  = Series.new({type: new Int32(), data: [1, 2, 3]});
+   * const df = new DataFrame({'a': a});
+   *
+   * const bc = await BlazingCluster.init();
+   * await bc.createTable('test_table', df);
+   * bc.describeTable('test_table'); // {'a': Int32}
+   * ```
+   */
+  describeTable(tableName: string): Map<string, DataType> {
+    return this.blazingContext.describeTable(tableName);
+  }
+
+  /**
+   * Returns a break down of a given query's logical relational algebra plan.
+   *
+   * @param sql SQL query
+   * @param detail if a physical plan should be returned instead
+   *
+   * @example
+   * ```typescript
+   * import {Series, DataFrame} from '@rapidsai/cudf';
+   * import {BlazingCluster} from '@rapidsai/blazingsql';
+   *
+   * const a  = Series.new([1, 2, 3]);
+   * const df = new DataFrame({'a': a});
+   *
+   * const bc = await BlazingCluster.init();
+   * await bc.createTable('test_table', df);
+   *
+   * bc.explain('SELECT a FROM test_table'); // BindableTableScan(table=[[main, test_table]],
+   * aliases=[[a]])
+   * ```
+   */
+  explain(sql: string, detail = false): string { return this.blazingContext.explain(sql, detail); }
+
+  /**
+   * Query a BlazingSQL table and return the result as a DataFrame.
+   *
+   * @param query SQL query string
+   *
+   * @example
+   * ```typescript
+   * import {Series, DataFrame, Int32} from '@rapidsai/cudf';
+   * import {BlazingCluster} from '@rapidsai/blazingsql';
+   *
+   * const a  = Series.new({type: new Int32(), data: [1, 2, 3]});
+   * const b  = Series.new({type: new Int32(), data: [4, 5, 6]});
+   * const df = new DataFrame({'a': a, 'b': b});
+   *
+   * const bc = await BlazingCluster.init();
+   * await bc.createTable('test_table', df);
+   *
+   * await bc.sql('SELECT a FROM test_table') // [1, 2, 3]
+   * ```
+   */
+  async sql(query: string): Promise<DataFrame> {
+    const queryPromises = [];
+
+    queryPromises.push(new Promise((resolve) => {
+      ctxToken++;
+      const messageId = this.generateMessageId(ctxToken);
+      setTimeout(() => {
+        const df = this.blazingContext.sql(query, ctxToken).result();
+        resolve({ctxToken: ctxToken, messageId, df});
+      });
+    }));
+
+    this.workers.forEach((worker) => {
+      queryPromises.push(new Promise((resolve) => {
+        ctxToken++;
+        const messageId = this.generateMessageId(ctxToken);
+        worker.send({operation: RUN_QUERY, ctxToken: ctxToken, messageId, query});
+        worker.once('message', (msg: any) => {
+          const {operation, ctxToken, messageId}: {
+            operation: string,
+            ctxToken: number,
+            messageId: string,
+          } = msg;
+
+          if (operation === QUERY_RAN) {
+            resolve({ctxToken, messageId, df: this.blazingContext.pullFromCache(messageId)});
+          }
+        });
+      }));
+    });
+
+    let result_df = new DataFrame({a: Series.new([])});
```

Comment:
```suggestion
    let result_df = new DataFrame({});
```

Reply:
https://github.com/rapidsai/node/issues/259
We need to keep it as is due to https://github.com/rapidsai/node/issues/259

Reply:
Why not just include the fix in this PR?

Reply:
No good reason, I guess the idea was to keep PRs pretty separate. I'll merge and include the fixes in this branch though.